pkg/cloud/azure/actuators/machineset/controller.go

```diff
@@ -132,30 +124,117 @@ func isInvalidConfigurationError(err error) bool {
 }
 
 func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, error) {
-	providerConfig, err := getproviderConfig(machineSet)
+	klog.Infof("%v: Reconciling MachineSet", machineSet.Name)
+	stockKeepUnit, err := getStockKeepUnit(r, machineSet)
 	if err != nil {
-		return ctrl.Result{}, mapierrors.InvalidMachineConfiguration("failed to get providerConfig: %v", err)
-	}
-	instanceType, ok := InstanceTypes[strings.ToLower(providerConfig.VMSize)]
-	if !ok {
-		klog.Errorf("Unable to set scale from zero annotations: unknown instance type: %s", providerConfig.VMSize)
-		klog.Errorf("Autoscaling from zero will not work. To fix this, manually populate machine annotations for your instance type: %v", []string{cpuKey, memoryKey, gpuKey})
+		if errors.Is(err, resourceskus.ErrResourceNotFound) {
+			// Print different error message when there is no failure, but SKU is not available.
+			klog.Errorf("Unable to set scale from zero annotations: instance type unknown or unavailabe for this account or location: %v", err)
+		} else {
+			klog.Errorf("Unable to set scale from zero annotations: Azure list SKU request failed: %v", err)
```

Comment:
Changed it to handle this better. Currently, it will retry only when it can't create a valid scope to fetch the SKU data.
Invalid machine spec configurations and successful Azure list-sku requests that don't find specified SKU's for VMSize won't retry.